syft/messaging/message.py

```diff
@@ -195,6 +200,114 @@ def detail(worker: AbstractWorker, msg_tuple: tuple) -> "Operation":
+        Args:
+            worker (AbstractWorker): a reference to the worker doing the serialization
+            ptr (Operation): a Message
+        Returns:
+            protobuf_obj: a Protobuf message holding the unique attributes of the message
+        Examples:
+            data = bufferize(message)
+        """
+        protobuf_op_msg = OperationMessagePB()
+        protobuf_op = OperationPB()
+        protobuf_op.command = operation.cmd_name
+
+        if type(operation.cmd_owner) == sy.generic.pointers.pointer_tensor.PointerTensor:
+            protobuf_owner = protobuf_op.owner_pointer
+        else:
+            protobuf_owner = protobuf_op.owner_tensor
+        protobuf_owner.CopyFrom(sy.serde.protobuf.serde._bufferize(worker, operation.cmd_owner))
+
+        if operation.cmd_args:
+            protobuf_op.args.extend(Operation._bufferize_args(worker, operation.cmd_args))
+
+        if operation.cmd_kwargs:
+            for key, value in operation.cmd_kwargs.items():
+                protobuf_op.kwargs.get_or_create(key).CopyFrom(
+                    Operation._bufferize_arg(worker, value)
+                )
+
+        for return_id in operation.return_ids:
+            sy.serde.protobuf.proto.set_protobuf_id(protobuf_op.return_ids.add(), return_id)
+
+        protobuf_op_msg.operation.CopyFrom(protobuf_op)
+        return protobuf_op_msg
+
+    @staticmethod
+    def unbufferize(worker: AbstractWorker, protobuf_obj: "OperationMessagePB") -> "Operation":
+        """
+        This function takes the Protobuf version of this message and converts
+        it into an Operation. The bufferize() method runs the inverse of this method.
+
+        Args:
+            worker (AbstractWorker): a reference to the worker necessary for detailing. Read
+                syft/serde/serde.py for more information on why this is necessary.
+            protobuf_obj (OperationPB): the Protobuf message
+
+        Returns:
+            obj (Operation): an Operation
+
+        Examples:
+            message = unbufferize(sy.local_worker, protobuf_msg)
+        """
+
+        command = protobuf_obj.operation.command
+        owner = sy.serde.protobuf.serde._unbufferize(
+            worker, getattr(protobuf_obj.operation, protobuf_obj.operation.WhichOneof("owner"))
+        )
+        args = Operation._unbufferize_args(worker, protobuf_obj.operation.args)
+
+        kwargs = {}
+        for key in protobuf_obj.operation.kwargs:
+            kwargs[key] = Operation._unbufferize_arg(worker, protobuf_obj.operation.kwargs[key])
+
+        return_ids = [
+            sy.serde.protobuf.proto.get_protobuf_id(pb_id)
+            for pb_id in protobuf_obj.operation.return_ids
+        ]
+
+        operation_msg = Operation(command, owner, tuple(args), kwargs, tuple(return_ids))
+
+        return operation_msg
+
+    @staticmethod
+    def _bufferize_args(worker: AbstractWorker, args: list) -> list:
```

Comment:
These look like general purpose methods that can be placed somewhere along with `get_protobuf_id`?

Reply:
Yeah, they maybe could be. Arg is only used in Operations so far. I don’t have a strong opinion either way.